File: python/vault_secrets.py
# python/vault_secrets.py
import hvac

def main():
    client = hvac.Client(url='http://localhost:8200', token='myroot')

    # Write a secret
    client.secrets.kv.v2.create_or_update_secret(
        path='mysecret',
        secret=dict(key='value')
    )

    # Read a secret
    secret = client.secrets.kv.v2.read_secret_version(path='mysecret')
    print(f"Secret: {secret['data']['data']['key']}")

if __name__ == "__main__":
    main()

File: examples/nodejs/vault_secrets.js
// nodejs/vault_secrets.js
const vault = require("node-vault")({
    apiVersion: "v1",
    endpoint: "http://localhost:8200",
    token: "myroot"
  });
  
  async function main() {
    try {
      // Write a secret
      await vault.write("secret/data/mysecret", { data: { key: "value" } });
  
      // Read a secret
      const result = await vault.read("secret/data/mysecret");
      console.log(`Secret: ${result.data.data.key}`);
    } catch (error) {
      console.error("Error:", error);
    }
  }
  
  main();